src/or/hs_service.c

```diff
@@ -1638,6 +1711,20 @@ build_service_desc_keys(const hs_service_t *service,
     ret = -1;
```

Comment:
Oh I think we want a `goto end` here, so we don't generate the emphemeral auth kp as well. For this, the label will need to be added. Same goes for a `goto` in the next if condition.